test/device/common.hpp

```diff
@@ -72,8 +72,21 @@ __global__ void fillKernel(DataType* data, uint32_t elementSize, uint32_t seed)
 
     if(index < elementSize)
     {
-        auto value  = (DataType(index / DataType(RAND_MAX) - 0.5) * 100) / elementSize;
-        data[index] = static_cast<DataType>(value);
+        if constexpr(std::is_same_v<DataType, hipFloatComplex>)
+        {
+            auto value  = (float(index / float(RAND_MAX) - 0.5) * 100) / elementSize;
+            data[index] = DataType(value, value);
```

Comment:
Since this check is explicitly for hipFloatComplex, then we should use the factory functions:
make_hipFloatComplex
make_hipDoubleComplex